### ISSUE TITLE
When there are residual directories in the cgroup, the log information is incorrect

### ISSUE THREAD

**Issue by yangjunmyfm192085** [kubernetes/kubernetes#111638]
### What happened?

In the kubernetes main branch, the following exception log appears in the kubelet log
```
E0729 01:14:33.037428  170935 manager.go:1123] Failed to create existing container: /system.slice/crio-ccc008b39f6ce2e2bfa0bbda7d8873d0164a7917a4c469ad2fbb5fc50ecb98b1.scope: Error finding container ccc008b39f6ce2e2bfa0bbda7d8873d0164a7917a4c469ad2fbb5fc50ecb98b1: Status 404 returned error &{%!s(*http.body=&{0xc0025f5bd8 <nil> <nil> false false {0 0} false false false <nil>}) {%!s(int32=0) %!s(uint32=0)} %!s(bool=false) <nil> %!s(func(error) error=0x835360) %!s(func() error=0x835460)}
```

The following log information output is wrong
return nil, fmt.Errorf("Error finding container %s: Status %d returned error %s", id, resp.StatusCode, resp.Body)

### What did you expect to happen?

the log information is as follows
```
E0729 02:59:50.722242  317325 manager.go:1123] Failed to create existing container: /system.slice/crio-eef4eeae4c31f001314dc155cb20012ddf3ff25400e0bfb097ccacac98006935.scope: Error finding container eef4eeae4c31f001314dc155cb20012ddf3ff25400e0bfb097ccacac98006935: Status 404 returned error can't find the container with id eef4eeae4c31f001314dc155cb20012ddf3ff25400e0bfb097ccacac98006935
```

### How can we reproduce it (as minimally and precisely as possible)?

cri is crio, and the path of the residual container exists in the cgroup directory

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
kubelet  --version=raw
version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.0-alpha.2.625+0532324bd17c58-dirty", GitCommit:"0532324bd17c58cd052da8c62a24ae8804167549", GitTreeState:"dirty", BuildDate:"2022-08-01T15:00:53Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>
curl -v --unix-socket /var/run/crio/crio.sock http://localhost/info | jq
*   Trying /var/run/crio/crio.sock:0...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to localhost (/var/run/crio/crio.sock) port 80 (#0)
> GET /info HTTP/1.1
> Host: localhost
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Tue, 02 Aug 2022 01:42:54 GMT
< Content-Length: 239
< 
{ [239 bytes data]
100   239  100   239    0     0   233k      0 --:--:-- --:--:-- --:--:--  233k
* Connection #0 to host localhost left intact
{
  "storage_driver": "overlay",
  "storage_root": "/var/lib/containers/storage",
  "cgroup_driver": "systemd",
  "default_id_mappings": {
    "uids": [
      {
        "container_id": 0,
        "host_id": 0,
        "size": 4294967295
      }
    ],
    "gids": [
      {
        "container_id": 0,
        "host_id": 0,
        "size": 4294967295
      }
    ]
  }
}
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by yangjunmyfm192085** [comment]
/cc @bobbypage , It looks like a bug of cAdvisor
the pr as follow try to fix it.
https://github.com/google/cadvisor/pull/3144

**Comment by bobbypage** [comment]
https://github.com/google/cadvisor/pull/3144 has been merged and will be included in cAdvisor v0.45.0 (k8s 1.25).
